File: GoonWorld/objects/DeathBox.cs
namespace GoonEngine.Objects;
using GoonEngine.Components;

public class DeathBox : GameObject
{
    protected PhysicsComponent _physicsComponent;
    protected DrawComponent _drawComponent;

    public DeathBox(object data) : base()
    {
        if (data is not TiledCS.TiledObject castedData)
            throw new Exception("Loading a player with no data somehow!");
        Location.X = (int)castedData.x;
        Location.Y = (int)castedData.y;
        _physicsComponent = new PhysicsComponent(new Models.BoundingBox(castedData.x, castedData.y, castedData.width, castedData.height)) { BodyType = (int)BodyTypes.DeathBox, GravityEnabled = false, StaticCollisionEnabled = false };
        AddComponent(_physicsComponent);
    }

    public override void Update()
    {
        base.Update();
    }
}

File: src/GoonWorld/gameobjects/DeathBox.cpp
#include <GoonWorld/gameobjects/DeathBox.hpp>
#include <GoonWorld/components/RigidbodyComponent.hpp>
#include <GoonEngine/rectangle.h>
using namespace GoonWorld;

DeathBox::DeathBox(TiledMap::TiledObject &object)
{
        auto bodyRect = geRectangle{object.X, object.Y, object.Width, object.Height};
        _rigidbodyComponent = new RigidbodyComponent(&bodyRect);
        _rigidbodyComponent->GravityEnabled(false);
        _rigidbodyComponent->SetBodyType(BodyTypes::DeathBox);
        AddComponent(_rigidbodyComponent);
}